### PR TITLE
Deprecated hpx::util::invoke_fused in favor of hpx::invoke_fused

### PR DESCRIPTION
**Pull request by dimitraka**
Deprecated:
-  `hpx::util::invoke_fused` in favor of `hpx::invoke_fused`
-  `hpx::util::invoke_fused_r` in favor of `hpx::invoke_fused_r`

Working towards resolving https://github.com/STEllAR-GROUP/hpx/issues/6000

### CODE REVIEW

File: libs/core/functional/docs/index.rst [hunk 1]
```diff
@@ -23,7 +23,7 @@ their arguments.
 * :cpp:func:`hpx::util::deferred_call`
 * :cpp:func:`hpx::invoke`
 * :cpp:func:`hpx::invoke_r`
-* :cpp:func:`hpx::util::invoke_fused`
+* :cpp:func:`hpx::invoke_fused`
```
Comment (hkaiser) [inline libs/core/functional/docs/index.rst:26]: Should we list `hpx::invoke_fused_r` here as well?

Comment (dimitraka) [reply libs/core/functional/docs/index.rst:26]: will do that!

File: docs/sphinx/api/public_api.rst [hunk 2]
```diff
@@ -479,7 +479,7 @@ Functions
    :cpp:func:`hpx::experimental::bind_back`
    :cpp:func:`hpx::bind_front`               :cppreference-generic:`utility/functional,bind_front`
    :cpp:func:`hpx::invoke`                   :cppreference-generic:`utility/functional,invoke`
-   :cpp:func:`hpx::util::invoke_fused`
+   :cpp:func:`hpx::invoke_fused`
```
Comment (hkaiser) [inline docs/sphinx/api/public_api.rst:482]: Should we add `invoke_fused_r` here as well (possibly in other places too)?

**Review by hkaiser — APPROVED** [review]
LGTM, thanks!

**Comment by hkaiser** [comment]
The remaining compilation errors have been fixed on master, let's get this merged.

**Comment by hkaiser** [comment]
bors merge

**Pull request merged**